### ISSUE TITLE
Fatal Error [2.0.1] in phalcon/mvc/model.zep:1322

### ISSUE THREAD

**Issue by salarmehr** [phalcon/cphalcon#10403]
This line of code in a Model validator  generates a fatal error:

 $model->appendMessage(new \Phalcon\Mvc\Model\Message('Error.', 'field));

````
PHP Fatal error:  Uncaught exception 'Exception' with message 'The argument is not initialized or 
iterable()' in phalcon/mvc/model.zep:1322
Fatal error: Uncaught exception 'Exception' with message 'The argument is not initialized or iterable()' in phalcon/mvc/model.zep:
````
Any idea? the error message point to zephir file!!!

**Comment by SidRoberts** [comment]
This is likely due to https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model.zep#L1319.

[Phalcon\Mvc\Model\Validator::getMessages()](https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model/validator.zep#L64) doesn't always return an array/iterator. The [_messages](https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model/validator.zep#L34) property is only turned into an array by the [Validator::appendMessage()](https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/mvc/model/validator.zep#L56) method which, presumably, isn't been called here.

**Comment by salarmehr** [comment]
@SidRoberts I did not get the point. Is it a bug? What should I do for that?

**Comment by SidRoberts** [comment]
It is a bug. I've submitted a [Pull Request](https://github.com/phalcon/cphalcon/pull/10404) fixing it. You can wait for it to be merged or you can rebuild Phalcon from this branch: https://github.com/SidRoberts/cphalcon/tree/fix-10403

**Comment by salarmehr** [comment]
@SidRoberts  Thanks for your response and pull request. It seems Phalcon v2 is on beta phase yet.

**Comment by andresgutierrez** [comment]
Fixed in 2.0.3
